pkg/tempopb/pool/pool_test.go

```diff
@@ -0,0 +1,74 @@
+// Forked with love from: https://github.com/prometheus/prometheus/tree/c954cd9d1d4e3530be2939d39d8633c38b70913f/util/pool
+
+package pool
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func makeFunc(size int) []byte {
+	return make([]byte, 0, size)
+}
+
+func TestPool(t *testing.T) {
+	testPool := New(1, 8, 2, makeFunc)
+	cases := []struct {
+		size        int
+		expectedCap int
+	}{
+		{
+			size:        -1,
+			expectedCap: 1,
+		},
+		{
+			size:        3,
+			expectedCap: 4,
+		},
+		{
+			size:        10,
+			expectedCap: 10,
+		},
+	}
+	for _, c := range cases {
+		ret := testPool.Get(c.size)
+		require.Equal(t, c.expectedCap, cap(ret))
+		testPool.Put(ret)
+	}
+}
+
+func TestPoolIgnoresRandomCapSlices(t *testing.T) {
```

Comment:
IIUC, this doesn't test if the pool ignores random cap sizes because of the `>=` in `require.True(t, cap(ret) >= size)`, but this test is still valuable.

Maybe we can add another test that tries to add a single buffer with a random (predetermined) size and checks if that exists in the pool?

Reply:
You can't really check if something exists in the pool. It's rather opaque. I think this code is about as close as you could get to what you're suggesting although we could do it more explicitly. I do like that this tests to make sure that the pool will not return a too small buffer if you put and get a bunch of random sized slices to/from it.

I'll change the name for clarity.

Reply:
Oh by "check if that exists in the pool" I meant just requesting back the same size buffer, but yeah fair